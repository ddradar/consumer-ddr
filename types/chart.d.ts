export type PlayStyle =
  | 'SINGLE'
  | 'DOUBLE'
  | 'COUPLE'
  | '6-PANELS'
  | '3-PANELS'
  | 'STEP BATTLE'

export type Chart = {
  songId: string
  playStyle: PlayStyle
  difficulty: number
  level: number | '10+'
}
