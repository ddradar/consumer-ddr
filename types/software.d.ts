type Region = 'JP' | 'US' | 'EU' | 'None'

type Platform =
  | 'Play Station'
  | 'Play Station 2'
  | 'Dreamcast'
  | 'GAME BOY COLOR'
  | 'Wii'

export type Software = {
  id: string
  name: string
  region: Region
  platform: Platform
  launched: Date | null
}
