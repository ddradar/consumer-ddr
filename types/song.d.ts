export type Song = {
  slug?: string
  series: string
  name: string
  artist: string
  bpm: number | string
  charts: Chart[]
}

export type Chart = {
  playStyle: PlayStyle
  difficulty: number
  level: number | '10+' | '?'
}

export type PlayStyle =
  | 'SINGLE'
  | 'DOUBLE'
  | 'COUPLE'
  | 'UNISON'
  | '6-PANELS'
  | '3-PANELS'
  | 'STEP BATTLE'
