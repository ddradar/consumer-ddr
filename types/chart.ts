type PlayStyle = 'SINGLE' | 'DOUBLE' | 'COUPLE' | '6-PANELS' | '3-PANELS'

export type Chart = {
  songId: string
  playStyle: PlayStyle
  difficulty: number
  level: number | '10+'
}
