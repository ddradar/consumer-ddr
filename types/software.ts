type Region = 'JP' | 'US' | 'EU'

type Platform = 'Play Station' | 'Play Station 2' | 'GAME BOY' | 'Wii'

export type Software = {
  id: string
  name: string
  region: Region
  platform: Platform
  launched: Date | null
}
