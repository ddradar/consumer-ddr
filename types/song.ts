export type Song = {
  slug?: string
  series: string
  name: string
  artist: string
  bpm: number | string
  charts: Chart[]
}

export type Chart = {
  playStyle: PlayStyle
  difficulty: number
  level: number | '10+' | '?'
}

export type PlayStyle =
  | 'SINGLE'
  | 'DOUBLE'
  | 'COUPLE'
  | 'UNISON'
  | '6-PANELS'
  | '3-PANELS'
  | 'STEP BATTLE'

export function normalizeDifficulty(difficultyName: string) {
  const name = difficultyName.toUpperCase()
  if (['BASIC', 'NORMAL', 'LIGHT'].includes(name)) return 'BASIC'
  if (['ANOTHER', 'DIFFICULT', 'TRICK'].includes(name)) return 'DIFFICULT'
  if (['MANIAC', 'SSR', 'EXPERT'].includes(name)) return 'EXPERT'
  if (['BEGINNER'].includes(name)) return 'BEGINNER'
  return 'CHALLENGE'
}
