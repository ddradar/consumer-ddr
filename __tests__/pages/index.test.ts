import { mount, RouterLinkStub, shallowMount, Wrapper } from '@vue/test-utils'

import { createVue } from '~/__tests__/utils'
import Index from '~/pages/index.vue'
import { Software } from '~/types/software'

type SoftListData = Omit<Software, 'difficultyNames' | 'region'> & {
  region: string
}

const localVue = createVue()

describe('pages/index.vue', () => {
  let wrapper: Wrapper<Index>
  const stubs = { NuxtLink: RouterLinkStub }
  beforeEach(() => {
    wrapper = shallowMount(Index, { localVue, stubs })
  })

  test('renders correctly', async () => {
    // Arrange - Act
    const softwareList: SoftListData[] = [
      {
        slug: '1st-jp',
        name: 'Dance Dance Revolution',
        platform: 'Play Station',
        region: '\u{1F1EF}\u{1F1F5}',
        launched: '1999-04-10'
      },
      {
        slug: '2nd-remix-jp',
        name: 'Dance Dance Revolution 2nd ReMIX',
        platform: 'Play Station',
        region: '\u{1F1EF}\u{1F1F5}',
        launched: '1999-08-26'
      }
    ]
    const data = () => ({ softwareList })
    const wrapper = mount(Index, { localVue, stubs, data })
    await wrapper.vm.$nextTick()

    // Assert
    expect(wrapper.element).toMatchSnapshot()
  })

  describe('head()', () => {
    test('returns { title: "Consumer DDR" }', () => {
      // Arrange
      const head = wrapper.vm.$options.head as any

      // Act & Assert
      expect(head.call(wrapper.vm)).toStrictEqual({
        title: 'Consumer DDR',
        titleTemplate: ''
      })
    })
  })
})
