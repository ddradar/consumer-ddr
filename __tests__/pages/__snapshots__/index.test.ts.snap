// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pages/index.vue > renders correctly 1`] = `
<section
  class="section"
>
  <h1
    class="title"
  >
    Consumer DDR
  </h1>
  <div
    class="b-table"
  >
    <div
      style="display: none;"
    >
      
      <span
        data-id="1"
      >
        Name
      </span>
      <span
        data-id="2"
      >
        Platform
      </span>
      <span
        data-id="3"
      >
        Launched
      </span>
      
    </div>
    <!--v-if-->
    <!--v-if-->
    <div
      class="table-wrapper"
    >
      <table
        class="table is-striped"
        tabindex="false"
      >
        <!--v-if-->
        <thead>
          <tr>
            <!--v-if-->
            <!--v-if-->
            
            <th
              class=""
              draggable="false"
            >
              <span>
                Name 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Platform 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            <th
              class=""
              draggable="false"
            >
              <span>
                Launched 
                <span
                  class="th-wrap sort-icon"
                  style="display: none;"
                >
                  <span
                    class="icon is-small"
                  >
                    <i
                      class="mdi mdi-arrow-up"
                    />
                  </span>
                </span>
              </span>
            </th>
            
            <!--v-if-->
          </tr>
          <!--v-if-->
          <!--v-if-->
        </thead>
        <tbody>
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="Name"
            >
              <a
                class="is-size-6-mobile"
              >
                Dance Dance Revolution (JP) 
              </a>
            </td>
            <td
              class=""
              data-label="Platform"
            >
              Play Station
            </td>
            <td
              class=""
              data-label="Launched"
            >
              1999-04-10
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub
            appear="false"
            css="true"
            name=""
            persisted="false"
          >
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <tr
            class=""
            draggable="false"
          >
            <!--v-if-->
            <!--v-if-->
            
            <td
              class=""
              data-label="Name"
            >
              <a
                class="is-size-6-mobile"
              >
                Dance Dance Revolution 2nd ReMIX (JP) 
              </a>
            </td>
            <td
              class=""
              data-label="Platform"
            >
              Play Station
            </td>
            <td
              class=""
              data-label="Launched"
            >
              1999-08-26
            </td>
            
            <!--v-if-->
          </tr>
          <transition-stub
            appear="false"
            css="true"
            name=""
            persisted="false"
          >
            <!--v-if-->
          </transition-stub>
          <!--v-if-->
          
          
          <!--v-if-->
        </tbody>
        <!--v-if-->
      </table>
      
      <transition-stub
        appear="false"
        css="true"
        name="fade"
        persisted="false"
      >
        <!--v-if-->
      </transition-stub>
      
    </div>
    <!--v-if-->
  </div>
</section>
`;
